pkg/monitor/api.go

```diff
@@ -203,6 +241,27 @@ func StartKubeAPIMonitoringWithConnectionReuse(ctx context.Context, m *Monitor,
 	return StartServerMonitoringWithConnectionReuse(ctx, m, clusterConfig, timeout, LocatorKubeAPIServerReusedConnection, "/api/v1/namespaces/default")
 }
 
+func StartKubeAPIMonitoringWithConnectionReuseForNodeIPs(ctx context.Context, m *Monitor, clusterConfig *rest.Config, timeout time.Duration) error {
+	client, err := kubernetes.NewForConfig(clusterConfig)
+	if err != nil {
+		return err
+	}
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(client, time.Second*30)
+	endpointsInformer := kubeInformerFactory.Core().V1().Endpoints()
```

Comment:
Watching all endpoints is expensive. Also, endpoints are going away soon.  I'd recommend you only watch in the default namespace